.github/workflows/ci.yml

```diff
@@ -33,26 +33,46 @@ jobs:
         run: sudo apt-get install uncrustify=0.69.0+dfsg1-1build1
       - name: Run Uncrustify
         run: |
+          # Uncrustify on C files while ignoring symlinks.
+          # Make a collapsible section in the log to run uncrustify
+          echo "::group::Uncrustify Check"
+          # GitHub automtically use "set -e" which causes scripts to fail on the first exit code
+          # This would mean the first time a file fails the check that we exit without formatting all files.
+          set +e
           uncrustify --version
-          find FreeRTOS/Demo/Common FreeRTOS/Test \( -name ethernet -o -name drivers -o -path 'FreeRTOS/Test/CMock/CMock' \) -prune -false -o -name "*.[hc]" -exec uncrustify --check -c tools/uncrustify.cfg {} +
+          find FreeRTOS/Demo/Common FreeRTOS/Test \( -name ethernet -o -name drivers -o -path 'FreeRTOS/Test/CMock/CMock' \) -prune -false -o -name "*.[hc]" | xargs uncrustify --no-backup --replace --if-changed -c tools/uncrustify.cfg -l C
+          echo "::endgroup::"
+          # Run a git diff to print the differences if any exist, return an error code if there are any
+          git diff --exit-code
+          if [ "$?" = "0" ]; then
+            echo -e "\033[32;3mUncrustify check passed\033[0m"
+            exit 0
+          else
+            echo -e "\033[32;31mFormatting check (using Uncrustify) failed...\033[0m"
+            # If there is an error, set this flag high again so the exit 1 fails the run
+            set -e
+            exit 1
+          fi
       - name: Check For Trailing Whitespace
         run: |
+          # Trailing Whitespace Check
           set +e
           ERROR=0
           find .  \( -name '.git' -o -path "./FreeRTOS/Test/CBMC/patches" -o -path "./FreeRTOS-Plus" -o -path "./FreeRTOS/Source"  -o  -path "./FreeRTOS/Test/CMock/CMock"  -o -path "./FreeRTOS/Demo" \) -prune -false -o -type f -a -name "*" -exec grep  -In -e "[[:blank:]]$" {} +
           if [ "$?" = "0" ]; then
-            echo "Files have trailing whitespace."
             ERROR=1
           fi
 
           find FreeRTOS/Demo/Common   \( -name "ethernet" \) -prune -o -false -o -type f -a -name "*" -exec grep --color=yes -In -e "[[:blank:]]$" {} +
           if [ "$?" = "0" ]; then
-            echo "Files have trailing whitespace."
+            echo -e "\033[32;31mFiles have trailing whitespace.\033[0m"
             exit 1
           else
             if [ "$ERROR" -eq "1" ]; then
+              echo -e "\033[32;31mFiles have trailing whitespace.\033[0m"
```

Comment:
I'm a little confused on why this echo was introduced? Wouldn't the echo statement on line 68 handle this case?

Reply:
This handles the case from like 61/62. It does two seperate checks for trailing whitespace. So line 68 prints if the second check hits a trailing whitespace character while this line prints if the first check hits a trailing whitespace character,

Reply:
Can we merge these two separate lines into one check like so:
```
if [ "$?" = "0" ] || [ "$ERROR" -eq "1" ]; then
```

That should work I believe.

Reply:
Yeah that would absolutely work. Since this has the two approvals already I'm going to merge it as it is. We know we want to update all these formatting actions in the near future where stuff like this could be fixed then.